spectral_cube/cube_utils.py

```diff
@@ -784,17 +795,65 @@ def combine_headers(header1, header2, **kwargs):
     wcs_opt, shape_opt = find_optimal_celestial_wcs([(s1, w1), (s2, w2)], auto_rotate=False,
                                                     **kwargs)
 
+    # find spectral coverage
+    specw1 = WCS(header1).spectral
+    specw2 = WCS(header2).spectral
+    specaxis1 = [x[0] for x in WCS(header1).world_axis_object_components].index('spectral')
+    specaxis2 = [x[0] for x in WCS(header2).world_axis_object_components].index('spectral')
+    range1 = specw1.pixel_to_world([0,header1[f'NAXIS{specaxis1+1}']-1])
+    range2 = specw2.pixel_to_world([0,header2[f'NAXIS{specaxis1+1}']-1])
+
+    # check for overlap
+    # this will raise an exception if the headers are an different units, which we want
+    if max(range1) < min(range2) or max(range2) < min(range1):
+        warnings.warn(f"There is no spectral overlap between {range1} and {range2}")
+
+    # check cdelt
+    dx1 = specw1.proj_plane_pixel_scales()[0]
+    dx2 = specw2.proj_plane_pixel_scales()[0]
+    if np.abs((dx1 - dx2)/dx1) > spectral_dx_threshold:
+        raise ValueError(f"Different spectral pixel scale {dx1} vs {dx2}")
+
+    ranges = np.hstack([range1, range2])
+    new_naxis = int(np.ceil((ranges.max() - ranges.min()) / np.abs(dx1)))
+    if specw1.wcs.cdelt == 1.0:
+        raise NotImplementedError("Spectral WCS doesn't have a CDELT parameter; we don't know how to deal with this generally.")
+    if np.sign(specw1.wcs.cdelt) == 1:
+        new_crval3 = ranges.min().to(u.Hz).value
```

Comment:
Keep in the same spectral unit?
```suggestion
        new_crval3 = ranges.min().value
```

Reply:
CRVAL has to be in Hz - astropy wcs (wcslib) will force it to be if it isn't

Reply:
I don't think this is the case, or only is when the spectral axis is already in freq. If it's velocity, the default is to m/s.

I added a couple commits that restructure this part to ensure it inherits the appropriate default WCS spectral unit (and this fixed our existing mosaic tests in this PR for me).